### ISSUE TITLE
Compatible static semver value and caret semver range fail to intersect

### ISSUE THREAD

**Issue by Silic0nS0ldier** [snyamathi/semver-intersect#1]
Attempting to perform...
```js
intersect('^1.0.0', '1.5.16');
```
...results in an error being throw that falsely states the ranges are not compatible.
```
Error: Range >=1.0.0 is not compatible with 1.5.16
```
It should resolve to `1.5.16`.

**Comment by Silic0nS0ldier** [comment]
Just done some quick debugging. Exception is throw after line 44 where there is a semver satisfaction check with the values `1.5.16` and `1.0.0`.

**Comment by snyamathi** [comment]
Thanks @Silic0nS0ldier, the changes in https://github.com/snyamathi/semver-intersect/commit/afbe59240ffc927ca4f97690705cf09111a79287 should fix the issue and has been published as version 1.1.1

```js
intersect('^1.0.0', '1.5.16'); // '1.5.16'
intersect('1.5.16', '^1.0.0'); // '1.5.16'
```
